src/app/ui/components/virtuoso.tsx

```diff
@@ -0,0 +1,45 @@
+import { ReactNode, useState } from 'react';
+
+import { Box } from 'leather-styles/jsx';
+
+import { useViewportMinWidth } from '@app/common/hooks/use-media-query';
+import { useOnResizeListener } from '@app/common/hooks/use-on-resize-listener';
+
+interface VirtuosoWrapper {
+  children: ReactNode;
+  hasFooter?: boolean;
+  isPopup?: boolean;
+}
+
+function vhToPixels(vh: string) {
+  const numericHeight = +vh.replace('vh', '');
+  return (numericHeight * window.innerHeight) / 100;
+}
+
+export function VirtuosoWrapper({ children, hasFooter, isPopup }: VirtuosoWrapper) {
```

Comment:
I should improve this more by:

- accepting the height as a prop
- using tokens for footer / header height
    - I couldn't get them working using `token('size`
    - When importing and using `token.size.X.value` it was giving type errors 

I can improve this further but I think this needs to get released sooner